src/Interpolation/CIC.hpp

```diff
@@ -63,7 +63,135 @@ namespace ippl {
+            Vector<T, Dim> to_in_grid_coordinates;
+            // Calculate the indices for the scatter operation
+            ippl::Vector<IndexType, Dim> fromi, toi;
+            for (unsigned int i = 0; i < Dim; i++) {
+                from[i] += hr[i] * T(0.5); //Centering offset
+                to[i]   += hr[i] * T(0.5); //Centering offset
+                from_in_grid_coordinates[i] = from[i] / hr[i];
+                to_in_grid_coordinates  [i] = to[i] / hr[i];
+                fromi[i] = floor(from_in_grid_coordinates[i]) + nghost;
+                toi[i]   = floor(to_in_grid_coordinates[i])   + nghost;
+            }
+            ippl::Vector<IndexType, Dim> fromi_local = fromi - lDom.first();
+            ippl::Vector<IndexType, Dim> toi_local = toi - lDom.first();
+            
+            // Calculate the relay point for each dimension
+            ippl::Vector<T, Dim> relay;
+            for (unsigned int i = 0; i < Dim; i++) {
+                relay[i] =
+                min(
+                    min(fromi[i] - nghost, toi[i] - nghost) * hr[i] + hr[i],
+                    max(
+                        max(fromi[i] - nghost, toi[i] - nghost) * hr[i],
+                        T(0.5) * (to[i] + from[i])
+                    )
+                );
+            }
+            
+            // Calculate jcfrom and jcto (deposited split currents)
+            ippl::Vector<T, Dim> jcfrom, jcto;
+            jcfrom = relay;
+            jcfrom -= from;
+            
+            jcto = to;
+            jcto -= relay;
+            
+            // Calculate wlo and whi
+            Vector<T, Dim> wlo, whi;
+            Vector<T, Dim> source1, source2;
+            for (unsigned i = 0; i < Dim; i++) {
+                wlo[i] = T(1.0) - fractional_part((from[i] + relay[i]) * T(0.5) / hr[i]);
+                whi[i] = fractional_part((from[i] + relay[i]) * T(0.5) / hr[i]);
+                source1[i] = (from[i] + relay[i]) * T(0.5);
+                source2[i] = (to[i]   + relay[i]) * T(0.5);
+            }
+            
+            // Perform the scatter operation for the first midpoint
+            auto _ =
+                (ZigzagScatterToPoint<ScatterPoint>(std::make_index_sequence<Dim>{}, view, wlo,
+                                                     whi, fromi_local, jcfrom, scale, hr, lDom, nghost, source1)
+                 ^ ...);
+            
+            for (unsigned i = 0; i < Dim; i++) {
+                wlo[i] = T(1.0) - fractional_part((to[i] + relay[i]) * T(0.5) / hr[i]);
+                whi[i] = fractional_part((to[i] + relay[i]) * T(0.5) / hr[i]);
+            }
 
+            auto __ =
+                (ZigzagScatterToPoint<ScatterPoint>(std::make_index_sequence<Dim>{}, view, wlo,
+                                                     whi, toi_local, jcto, scale, hr, lDom, nghost, source2)
+                 ^ ...);
+
+            (void)_;
+            (void)__; // [[maybe_unused]] causes issues on certain compilers
```

Comment:
Does the variable appear unused if you reassign the first placeholder instead of creating a new one?

Reply:
to my surprise the warning disappears, I renamed to dummy_ since double underscore lead are reserved anyway